File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.cs
using System;
using System.Text;
using System.Linq;
public class Solution {
    public string solution(int[] food) {
        string answer = "";
        StringBuilder sb = new StringBuilder(100);
        for(int i=1;i<food.Length;++i)
        {
            int foodCount = (int)food[i]/2;
            for(int j=0;j<foodCount;++j)
            {
                sb.Append(i);
            }
        }
        string tmp = new string(sb.ToString().Reverse().ToArray());
        sb.Append(0).Append(tmp);
        answer = sb.ToString();
        return answer;
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution2(vector<int> food) {
    string answer = "";
    string temp{};    
    
    for(int i = 1 ;i<food.size(); i++)
    {
        int place = food[i]/2;
        for(int j=0; j < place; j++)
        {
            answer += i + '0';
        }
    }
    temp = answer;
    reverse(temp.begin(),temp.end());
    
    answer.append("0").append(temp);
    
    return answer;
}

string solution(vector<int> food) {
    string answer = "";
    string temp{};

    auto it = food.begin();
    it++;
    for(auto it = food.begin(); it != food.end(); ++it)
    {
        int num = (*it)/2;
        
        for(int j=0; j<num;++j)
        {
            answer += (char)(it-food.begin())+'0';
        }
    }
    
    temp = answer;
    reverse(temp.begin(),temp.end());
    
    answer.append("0").append(temp);
    
        
    
    return answer;
}




// 포문 1번 부터 시작
// 2로 나눈 몫만 사용해서 문자열에 넣고
// 반복
// 마지막에 0과 reverse한 내용을 넣어서 마무리